notify-debouncer-full/src/lib.rs

```diff
@@ -713,6 +703,39 @@ pub fn new_debouncer<F: DebounceEventHandler>(
     )
 }
 
+fn sort_events(events: Vec<DebouncedEvent>) -> Vec<DebouncedEvent> {
+    let mut sorted = Vec::with_capacity(events.len());
+
+    // group events by path
+    let mut events_by_path: HashMap<_, VecDeque<_>> =
+        events.into_iter().fold(HashMap::new(), |mut acc, event| {
+            acc.entry(event.paths.last().cloned().unwrap_or_default())
+                .or_default()
+                .push_back(event);
+            acc
+        });
+
+    // push events for different paths in chronological order and keep the order of events with the same path
+    while !events_by_path.is_empty() {
+        let min_time = events_by_path
+            .values()
+            .map(|events| events[0].time)
+            .min()
```

Comment:
Seem with that total algorithmic complexity has O(n^2). May be it is need to use binary heap or something like this to track min_time?

Reply:
Using `BinaryHeap` improves performance a bit. Especially in scenarios with many files.

I didn't commit the benchmark because I don't know if it's useful enough. But here are the results. Unfortunately they are a bit over the place, so I'm not sure how reliable they are:

```
sort-events/sort_events with 10 files and 100k events each
    new:   [328.05 ms 330.98 ms 334.31 ms]
    old:   [408.45 ms 453.59 ms 512.18 ms] +37.042%

sort-events/sort_events with 1k files and 1k events each
    new:   [398.06 ms 401.66 ms 405.87 ms]
    old:   [448.35 ms 454.64 ms 462.24 ms] +13.190%

sort-events/sort_events with 100k files and 10 events each
    new:   [841.49 ms 906.83 ms 1.0303 s]
    old:   aborted after 10 minutes
```